src/courses/mappers/18.md

```diff
@@ -0,0 +1,271 @@
+---
+order: 18
+title: Appendix - SAF CLI Integration
+author: Charles Hu
+---
+
+## SAF CLI Integration
+
+The following is a supplemental lesson on how to integrate your developed mapper with the SAF CLI. Integration with the SAF CLI will allow your mapper to be used in a command line environment, independent of Heimdall2.
+
+If you have not yet created a mapper, please follow the primary course and do so. Once completed, you will be able to continue with this lesson.
+
+### Set Up
+
+First, we need set up the necessary files to begin integrating your mapper with the SAF CLI.
+
+1. Create a development branch against the [SAF CLI repository](https://github.com/mitre/saf) and create a draft pull request for your new branch.
+
+2. In the `package.json` file, update the versions of `@mitre/hdf-converters` and `@mitre/heimdall-lite` to the [latest release of Heimdall2](https://github.com/mitre/heimdall2/releases).
+
+3. In the `src/commands/convert` directory, create a blank TypeScript file. It should be named:
+```
+{YOUR-EXPORT-NAME-HERE}2hdf.ts
+```
+
+4. In the `test/sample_data` directory, create a directory named `{YOUR-EXPORT-NAME-HERE}`. Underneath it, create a directory named `sample_input_report`. The file structure should now look like this:
+```
++-- sample_data
+|   +-- {YOUR-EXPORT-NAME-HERE}
+|   |   +-- sample_input_report
+```
+
+5. Place your sample export under the `sample_input_report` directory. Your sample export should be genericized to avoid any leaking of sensitive information. Under the `{YOUR-EXPORT-NAME-HERE}` directory, place your output OHDF files generated during the original testing phase of your mapper development. The file structure should now look like this:
+```
++-- sample_data
+|   +-- {YOUR-EXPORT-NAME-HERE}
+|   |   +-- sample_input_report
+|   |   |   +-- {YOUR-SAMPLE-EXPORT}
+|   |   +-- {YOUR-EXPORT-NAME-HERE}-hdf.json
+|   |   +-- {YOUR-EXPORT-NAME-HERE}-hdf-withraw.json
+```
+
+6. In the `test/commands/convert` directory, create a blank TypeScript file. It should be named:
+```
+{YOUR-EXPORT-NAME-HERE}2hdf.test.ts
+```
+
+### Integration
```

Comment:
might be worthwhile to have the reminder section here about how it might be necessary to extract data from an api of some kind in which case to reach out to us or look at the api based examples that we have

Reply:
Addressed.